packages/typespec-client-generator-core/src/media-types.ts

```diff
@@ -0,0 +1,105 @@
+const json = "json";
+const xml = "xml";
+const application = "application";
+const text = "text";
+const multipart = "multipart";
+const formData = "form-data";
+const formEncoded = "x-www-form-urlencoded";
+
+export enum KnownMediaType {
+  Json = "json",
+  Xml = "xml",
+  Form = "form",
+  Binary = "binary",
+  Multipart = "multipart",
+  Text = "text",
+
+  Unknown = "unknown",
+}
+
+export enum FormatType {
+  QueryParameter = "-query-parameter-",
+  UriParameter = "-uri-parameter-",
+
+  Header = "-header-",
+  Cookie = "-cookie-",
+}
+
+export function parseMediaType(mediaType: string) {
+  if (mediaType) {
+    const parsed =
+      /(application|audio|font|example|image|message|model|multipart|text|video|x-(?:[0-9A-Za-z!#$%&'*+.^_`|~-]+))\/([0-9A-Za-z!#$%&'*.^_`|~-]+)\s*(?:\+([0-9A-Za-z!#$%&'*.^_`|~-]+))?\s*(?:;.\s*(\S*))?/g.exec(
+        mediaType,
+      );
+    if (parsed) {
+      return {
+        type: parsed[1],
+        subtype: parsed[2],
+        suffix: parsed[3],
+        parameter: parsed[4],
+      };
+    }
+  }
+  return undefined;
+}
+
+export function knownMediaType(mediaType: string) {
```

Comment:
should this be `getKnownMediaType` or something? this method name is missing a verb

Reply:
also I think we should put the explicit return type here.

Reply:
code is copied from autorest [impl](https://github.com/Azure/autorest/blob/faf5c1168232ba8a1e8fe02fbc28667c00db8c96/packages/libs/codegen/src/media-types.ts#L53). let me remove the useless code for now.